### ISSUE TITLE
[bug] no-missing-key does not check ConditionalExpression / LogicalExpression inside return statements of .map callbacks with block bodies

### ISSUE THREAD

**Issue by lixiaoyan** [Rel1cx/eslint-react#1766]
### Describe the bug

The `no-missing-key` rule correctly handles ternary and logical expressions when the `.map` (or `Array.from`) callback uses a concise arrow body, but **fails to handle them when the callback uses a block body with `return`**.

Looking at the source in `plugins/eslint-plugin-react-x/src/rules/no-missing-key/no-missing-key.ts`:

For concise arrow bodies, `checkExpr` is called, which unwraps `ConditionalExpression` (and `LogicalExpression`) and runs `check` on each branch:

```ts
function checkExpr(node: TSESTree.Expression): Descriptor | null {
  switch (node.type) {
    case AST.ConditionalExpression:
      return check(node.consequent) ?? check(node.alternate);
    case AST.LogicalExpression:
      return check(node.left) ?? check(node.right);
    case AST.JSXElement:
    case AST.JSXFragment:
      return check(node);
    default:
      return null;
  }
}
```

For block bodies, `checkBlock` is called, which iterates `ReturnStatement`s and passes the argument directly to `check`:

```ts
function checkBlock(node: TSESTree.BlockStatement): Descriptor[] {
  const descriptors: Descriptor[] = [];
  for (const stmt of getNestedReturnStatements(node)) {
    if (stmt.argument == null) continue;
    const desc = check(stmt.argument);
    if (desc == null) continue;
    descriptors.push(desc);
  }
  return descriptors;
}
```

But `check` only matches `JSXElement` and `JSXFragment` — it does not recurse into `ConditionalExpression` or `LogicalExpression`. So a `return cond ? <A/> : <B/>` inside a block body is silently ignored. This is inconsistent with the concise-body path and produces false negatives.

### Reproduction

```jsx
const items = [1, 2, 3];

// ✅ Correctly reported (concise body)
items.map((x) => (cond ? <div /> : <span />));

// ❌ NOT reported — false negative (block body + return)
items.map((x) => {
  return cond ? <div /> : <span />;
});

// Same issue with logical expressions
items.map((x) => {
  return cond && <div />;
});
```

### Expected behavior

Both forms should report missing `key` on the JSX elements in each branch. The `return`-statement path should also unwrap `ConditionalExpression` and `LogicalExpression` — ideally by reusing `checkExpr` on `stmt.argument`:

```ts
const desc = checkExpr(stmt.argument);
```

(Possibly separate enhancement: `checkExpr` short-circuits with `??`, so when both branches of a ternary lack a key only one is reported. It might be better to report both.)

### Platform and versions

```Text
eslint: 10.3.0
typescript-eslint: 8.59.3
@eslint-react/eslint-plugin: 5.7.7
```

### Stack trace

```Text

```

**Comment by lixiaoyan** [comment]
For reference: Biome (`useJsxKeyInIterable` 2.4.15) and Oxlint (`react/jsx-key` 1.64.0) both handle these cases correctly, and `eslint-plugin-react` has already fixed this on `main` (not yet released as of 7.37.5).

- https://github.com/jsx-eslint/eslint-plugin-react/commit/a116665c81f89b874dd374842f673720ec18bc38
- https://github.com/jsx-eslint/eslint-plugin-react/commit/36b8e439ad9ac62d2ec0e258943a1508e6a4cd1a
